ext/session/session.c

```diff
@@ -2659,15 +2658,29 @@ PHP_FUNCTION(session_start)
 
 	/* set options */
 	if (options) {
-		ZEND_HASH_FOREACH_KEY_VAL(Z_ARRVAL_P(options), num_idx, str_idx, value) {
+		if (UNEXPECTED(HT_IS_PACKED(Z_ARRVAL_P(options)))) {
+		}
+
+		ZEND_HASH_FOREACH_STR_KEY_VAL(Z_ARRVAL_P(options), str_idx, value) {
 			if (str_idx) {
 				switch(Z_TYPE_P(value)) {
```

Comment:
Maybe swap the conditions around and do `if (UNEXPECTED(str_index == NULL)) { /* throw error */ }` and then you don't need to indent the happy path.